build_posix/Make.subdirs

```diff
@@ -20,8 +21,8 @@ ext/test/kvs_bdb HAVE_BERKELEY_DB
 api/leveldb LEVELDB
 bench/wtperf
 examples/c
-lang/java JAVA
 examples/java JAVA
+lang/java JAVA
```

Comment:
The ordering here matters: we can't build the Java examples until the Java language support is built.